### PR TITLE
authz: add e2e tests for negative match and wildcard

### PR DESCRIPTION
**Pull request by yangminzhu**
For https://github.com/istio/istio/issues/19894

This PR adds e2e tests for the negative match (`not_` fields) and the wildcard in authz policy including the following test cases:
- overlap cases when `paths` and `not_paths` are used together
- deny traffic from other namespaces, in other words, only allow same namespace traffic
- deny plain text traffic, in other words, only allow mTLS traffic

### CODE REVIEW

File: tests/integration/security/testdata/rbac/v1beta1-negative-match.yaml.tmpl [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+# The following policy enables mTLS with PERMISSIVE mode for workload b, c and d
+
+apiVersion: authentication.istio.io/v1alpha1
+kind: Policy
+metadata:
+  name: "permissive"
+  namespace: "{{ .Namespace }}"
+spec:
+  targets:
+  - name: b
```
Comment (incfly) [inline tests/integration/security/testdata/rbac/v1beta1-negative-match.yaml.tmpl]: do we have to use targets? This is about to be deprecated. @diemtvu

Comment (incfly) [reply tests/integration/security/testdata/rbac/v1beta1-negative-match.yaml.tmpl]: can we construct tests to avoid referencing this field?

**Comment by howardjohn** [comment]
(my approval was for T&R part, I didn't look closely at the authz stuff)

File: tests/integration/security/rbac_v1beta1_test.go [hunk 2]
```diff
@@ -352,3 +352,73 @@ func TestV1beta1_Deny(t *testing.T) {
...
+				Inject: true,
+			})
+			ns2 := namespace.NewOrFail(t, ctx, namespace.Config{
+				Prefix: "v1beta1-negative-match-2",
+				Inject: true,
+			})
+
+			args := map[string]string{
+				"Namespace":  ns.Name(),
+				"Namespace2": ns2.Name(),
```
Comment (incfly) [inline tests/integration/security/rbac_v1beta1_test.go]: can we give it a more informative name other tha ns2? sth like mtlsNS?

File: tests/integration/security/rbac_v1beta1_test.go [hunk 3]
```diff
@@ -352,3 +352,73 @@ func TestV1beta1_Deny(t *testing.T) {
...
+							Target:   target,
+							PortName: "http",
+							Scheme:   scheme.HTTP,
+							Path:     path,
+						},
+					},
+					ExpectAllowed: expectAllowed,
+				}
+			}
+			cases := []rbacUtil.TestCase{
```
Comment (incfly) [inline tests/integration/security/rbac_v1beta1_test.go]: optional, would be better to have a paragraph of overall setup and then per test case oneline comment to explain why each test case expect allow/deny.

otherwise, we need to follow the yaml CRD and understand bottom-up.

File: tests/integration/security/testdata/rbac/v1beta1-negative-match.yaml.tmpl (hunk 1 above)
Comment (yangminzhu) [reply tests/integration/security/testdata/rbac/v1beta1-negative-match.yaml.tmpl]: done

File: tests/integration/security/rbac_v1beta1_test.go (hunk 2 above)
Comment (yangminzhu) [reply tests/integration/security/rbac_v1beta1_test.go]: that doesn't make too much sense, mtlsNS is inaccurate here. we just need 2 different namespaces.

File: tests/integration/security/rbac_v1beta1_test.go (hunk 3 above)
Comment (yangminzhu) [reply tests/integration/security/rbac_v1beta1_test.go]: added more comments.

File: tests/integration/security/testdata/rbac/v1beta1-negative-match.yaml.tmpl [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  action: DENY
+  rules:
+  - from:
+    - source:
+        notNamespaces: ["{{ .Namespace }}"]
+---
+
+# The following policy denies access to workload d if it's not mTLS, in other words,
+# it allows only mTLS traffic to access workload d
+
```
Comment (liminw) [inline tests/integration/security/testdata/rbac/v1beta1-negative-match.yaml.tmpl]: As discussed, this example should be documented in official istio.io page.

Comment (yangminzhu) [reply tests/integration/security/testdata/rbac/v1beta1-negative-match.yaml.tmpl]: agree, I will send out the doc update to rigs.
